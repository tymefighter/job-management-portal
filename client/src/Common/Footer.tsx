import { Link } from "react-router-dom";

import "../styles/Footer.scss";

function Footer() {
    return (
        <footer className="footer">
            <div className="footer__section">
                <h2 className="footer__section-head">Nav</h2>
                <ul className="footer__section-list">
                    <li><Link to="/">Home</Link></li>
                    <li><Link to="/companies">Company</Link></li>
                    <li><Link to="/jobs">Jobs</Link></li>
                    <li><Link to="/information">Information</Link></li>
                </ul>
            </div>
            <div className="footer__section">
                <h2 className="footer__section-head">Contact Me</h2>
                <ul className="footer__section-list">
                    <li> Email me at{" "}
                        <a href="mailto:ahmed.dadarkar@gmail.com">
                            ahmed.dadarkar@gmail.com
                        </a>
                    </li>
                    <li>Phone Number: +91 7594069315</li>
                    <li>WhatsApp Phone Number: +91 9920460903</li>
                </ul>
            </div>
            <div className="footer__section">
                <h2 className="footer__section-head">Ahmed Zaheer Dadarkar</h2>
                <p className="footer__section-para">
                    I am <b>Ahmed Zaheer Dadarkar</b>, also known as <b>tymefighter</b>.
                    I've built this website to practice my HTML, CSS, JS, React, Redux
                    and Typescript skills. This website allows one to manage companies
                    and jobs through an accessible user interface. Hope you enjoy the
                    user experience while engaging with this application !
                </p>
            </div>
        </footer>
    );
}

export default Footer;