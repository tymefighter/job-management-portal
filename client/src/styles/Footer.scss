.footer {
    display: flex;
    padding: 0 2rem;
}

.footer__section {
    flex: 1 30rem;
}

.footer__section-head {
    font-size: 1.5rem;
}

.footer__section-list {
    list-style: none;
    padding: 0;
    font-size: 1.2rem;
}

.footer__section-para {
    font-size: 1.2rem;
}