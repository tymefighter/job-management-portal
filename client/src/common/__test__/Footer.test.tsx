import { BrowserRouter, Switch, Route } from "react-router-dom";
import { render } from "@testing-library/react";
import Footer from "../Footer";

beforeEach(() => {
    window.history.pushState(undefined, "", "/");
});

test("Footer Snapshot test", () => {
    const element = (
        <BrowserRouter>
            <div id="very-strange-test-id">
                <Footer />
            </div>
        </BrowserRouter>
    );

    const {container} = render(element);

    expect(container.querySelector("#very-strange-test-id"))
        .toMatchSnapshot("Footer Snapshot");
});

test("Footer Link test by clicking", () => {
    const companiesValue = "C_O_M_P_A_N_I_E_S";
    const jobsValue = "J_O_B_S";
    const infoValue = "I_N_F_O";
    const homeValue = "H_O_M_E";

    const element = (
        <BrowserRouter>
            <Switch>
                <Route path="/companies">{companiesValue}</Route>
                <Route path="/jobs">{jobsValue}</Route>
                <Route path="/information">{infoValue}</Route>
                <Route path="/">{homeValue}</Route>
            </Switch>
            <Footer />
        </BrowserRouter>
    );

    const {container, getByText} = render(element);

    expect(window.location.pathname).toEqual("/");
    expect(container.textContent).toContain(homeValue);

    const companiesLink = getByText("Company");
    companiesLink.click();
    expect(window.location.pathname).toEqual("/companies");
    expect(container.textContent).toContain(companiesValue);

    const jobsLink = getByText("Jobs");
    jobsLink.click();
    expect(window.location.pathname).toEqual("/jobs");
    expect(container.textContent).toContain(jobsValue);

    const infoLink = getByText("Information");
    infoLink.click();
    expect(window.location.pathname).toEqual("/information");
    expect(container.textContent).toContain(infoValue);

    const homeLink = getByText("Home");
    homeLink.click();
    expect(window.location.pathname).toEqual("/");
    expect(container.textContent).toContain(homeValue);
});