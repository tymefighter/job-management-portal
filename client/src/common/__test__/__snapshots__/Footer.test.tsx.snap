// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot test: Footer Snapshot 1`] = `
<div
  id="very-strange-test-id"
>
  <footer
    class="footer"
  >
    <div
      class="footer__nav footer__section"
    >
      <h2
        class="footer__section-head"
      >
        Nav
      </h2>
      <ul
        class="footer__section-list"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/companies"
          >
            Company
          </a>
        </li>
        <li>
          <a
            href="/jobs"
          >
            Jobs
          </a>
        </li>
        <li>
          <a
            href="/information"
          >
            Information
          </a>
        </li>
      </ul>
    </div>
    <div
      class="footer__contact footer__section"
    >
      <h2
        class="footer__section-head"
      >
        Contact Me
      </h2>
      <ul
        class="footer__section-list"
      >
        <li>
           Email me at
           
          <a
            href="mailto:ahmed.dadarkar@gmail.com"
          >
            ahmed.dadarkar@gmail.com
          </a>
        </li>
        <li>
          Phone Number: +91 7594069315
        </li>
        <li>
          WhatsApp Phone Number: +91 9920460903
        </li>
      </ul>
    </div>
    <div
      class="footer__me footer__section"
    >
      <h2
        class="footer__section-head"
      >
        Ahmed Zaheer Dadarkar
      </h2>
      <p
        class="footer__section-para"
      >
        I am 
        <b>
          Ahmed Zaheer Dadarkar
        </b>
        , also known as 
        <b>
          tymefighter
        </b>
        . I've built this website to practice my HTML, CSS, JS, React, Redux and Typescript skills. This website allows one to manage companies and jobs through an accessible user interface. Hope you enjoy the user experience while engaging with this application !
      </p>
    </div>
  </footer>
</div>
`;
